### ISSUE TITLE
Implement accessors (duk_def_prop)

### ISSUE THREAD

**Issue by wyozi** [SkylerLipthay/ducc#11]
Hello again!

Some code I'm writing would benefit from [duk_def_prop](https://duktape.org/api.html#duk_def_prop), i.e. implementing properties with getters and setters straight in Rust. It's doable by calling `defineProperty` in JS, but doing it in Rust might feel a bit nicer.

I would be open to implementing the feature, but I'm not sure what the API should look like since `duk_def_prop` accepts quite a varying set of arguments.

**Comment by SkylerLipthay** [comment]
How about something like this?

```rust
pub fn define<K: ToValue<'ducc>>(&self, key: K, desc: PropertyDescriptor<'ducc>) -> Result<()> { ... }

struct PropertyDescriptor<'ducc> {
    enumerable: bool,
    configurable: bool,
    writable: bool,
    value: Option<Value<'ducc>>,
    get: Option<Function<'ducc>>,
    set: Option<Function<'ducc>>,
}

// Optionally hook up with `duk_get_prop_desc`:
// pub fn definition<K: ToValue<'ducc>>(&self, key: K) -> Result<PropertyDescriptor<'ducc>> { ... }
```

`PropertyDescriptor` should probably implement the [builder pattern](https://doc.rust-lang.org/1.0.0/style/ownership/builders.html). `DUK_DEFPROP_FORCE` is available but isn't part of the descriptor metadata per se. It could be a function parameter like `Object::define(&self, key, desc, force: bool)` or it could just be never/always set in the inner `duk_def_prop` call. (Depends on your needs. ducc only implements what its users need it to do, it doesn't aim to preemptively cover all use cases.)

My plate is full with other work right now but I'd happily review/advise on any work you do here!

**Comment by wyozi** [comment]
Okay, created preliminary PR at #12

**Comment by SkylerLipthay** [comment]
Closed by #12. A new version `ducc 0.1.5` has been published.

Thanks for your contributions!!
